js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            //             "commission": "0.016798204",
+            //             "commissionAsset": "USDT",
+            //             "time": "1647718055000",
+            //             "isBuyer": true,
+            //             "isMaker": false,
+            //             "isBestMatch": true
+            //         }
+            //     ]
+            //
+        } else {
+            request['order_id'] = id;
+            const response = await this.contractPrivateGetOrderDealDetailsOrderId (this.extend (request, query));
+            //
+            //     {
+            //         "success": true,
+            //         "code": "0",
+            //         "data": [
+            //             {
+            //                 "id": "299444585",
+            //                 "symbol": "STEPN_USDT",
+            //                 "side": "1",
+            //                 "vol": "1",
+            //                 "price": "2.45455",
+            //                 "feeCurrency": "USDT",
+            //                 "fee": "0.00147273",
+            //                 "timestamp": "1648924557000",
+            //                 "profit": "0",
+            //                 "category": "1",
+            //                 "orderId": "265307163526610432",
+            //                 "positionMode": "1",
+            //                 "taker": true
+            //             }
+            //         ]
+            //     }
+            //
+            trades = this.safeValue (response, 'data');
+        }
+        return this.parseTrades (trades, market, since, limit, query);
+    }
+
+    async modifyMarginHelper (symbol, amount, addOrReduce, params = {}) {
+        const positionId = this.safeInteger (params, 'positionId');
+        if (positionId === undefined) {
+            throw new ArgumentsRequired (this.id + ' modifyMarginHelper() requires a positionId parameter');
+        }
+        await this.loadMarkets ();
+        const request = {
+            'positionId': positionId,
+            'amount': amount,
+            'type': addOrReduce,
+        };
+        const response = await this.contractPrivatePostPositionChangeMargin (this.extend (request, params));
+        //
+        //     {
+        //         "success": true,
+        //         "code": 0
+        //     }
+        return response;
+    }
+
+    async reduceMargin (symbol, amount, params = {}) {
+        return await this.modifyMarginHelper (symbol, amount, 'SUB', params);
+    }
+
+    async addMargin (symbol, amount, params = {}) {
+        return await this.modifyMarginHelper (symbol, amount, 'ADD', params);
+    }
+
+    async setLeverage (leverage, symbol = undefined, params = {}) {
+        const positionId = this.safeInteger (params, 'positionId');
+        if (positionId === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a positionId parameter');
```

Comment:
Can't we choose between providing a positionId or openType+symbol+positionType?
![image](https://user-images.githubusercontent.com/43336371/164754068-6cc5a310-3c9a-4290-87a3-49cf8a6cb5ea.png)

Reply:
done